### ISSUE TITLE
Consider merging mutliple `isinstance()` calls together

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#603]
# Rule request

## Thesis

One can write code like: 

```python
if isinstance(some, int) or isinstance(some, str):
    ...
```

When we have more than one `isinstance` call with the same argument and `or` condition - we can squash it to `isinstance(some, (int, str))`

## Reasoning

It is more compact and more readable.
Original: https://github.com/wemake-services/wemake-python-styleguide/issues/431
Related: https://github.com/PyCQA/pylint/blob/5a86fddcb815ebd08fbc3ffcbaf7b11c2ba2e107/pylint/checkers/refactoring.py#L686

**Comment by DjaPy** [comment]
Hi. Sorry of my English.

I have questions:

1. Do I understand correctly, that the error number is 431?
2. In the beginning you need to create a `violation`, after `visitor`?
3. How can I call a violation? Something like `MergingIsinstanceViolations`?
4. How can a link to pylint help me?
5. When writing visitor it is better to inherit from BaseNodeVisitor?

Sorry about the noob questions.

**Comment by sobolevn** [comment]
1. Kind of. It should be `4xx` but the exact number is not important. Just use `previous + 1`
2. Order of `violation` / `visitor` is not important. You need to create both as a result. I would recommend to start with a violation
3. You can call it `ConsecutiveTypeCheckViolation`
4. It has some inspiration to the code you need. Our `ast` implementation is different (they use `astor`, we use `ast`), but it can give you a hint
5. Yes, totally

All questions are welcome! Your help is appreciated and your work is improving the project. 👍 
Thank you!

**Comment by elkku26** [comment]
Hi, I'd love to help out. I'm a bit new and I was wondering where I could find the code in question here. Thanks in advance!

**Comment by sobolevn** [comment]
What do you mean by
> where I could find the code in question here

?

**Comment by elkku26** [comment]
@sobolevn I think I figured it out. If I'm not completely lost I think I'm supposed to find the kinds of redundant if-statements you mentioned in the refactoring.py and fix them, right?

**Comment by sobolevn** [comment]
@elkku26 I guess, that you have completely missed the point of this library.
We are writing a linter - a tool to find bugs in your code - and this task is to create a new rule for this linter.

So, the goal is to write some check that will find bad code in other people's projects. 
Not to find it here.

**Comment by sobolevn** [comment]
@DjaPy and @elkku26 how's your progress? I can take this over.
I am planning a release pretty soon. And I want to have all compare related stuff in there.
